### DISCUSSION TITLE
Reduce RAM usage

### DISCUSSION THREAD

**Discussion by Orange-Murker** [pretix/pretix#3632] [Q&A] [answered]
I am running Pretix in a docker container with Redis and what I see is that the pretix container is using 10.67GiB of RAM.
From looking at processes running in the container it seems like celery and gunicorn are causing this.

Here is a part of the config responsible for redis and celery:
```
[redis]
location=redis://redis:6379/0
sessions=true

[celery]
backend=redis://redis:6379/0
broker=redis://redis:6379/0
```

What are the steps that I can take to investigate this further and reduce RAM usage?

**Comment by raphaelm** [comment, marked as answer]
You can reduce the number of threads, `--workers` in gunicorn and `--concurrency` in celery. The docker container defaults to setting both to the number of CPU cores, which is usually a good tradeoff, but if you have a machine with many cores and little RAM it might be problematic. RAM usage scales pretty linearly with these values.

Other than that there's not much you can do.

**Comment by mklemme1** [comment]
Hallo  @raphaelm 

Where would I set the two options in a Docker environment?
I am using the standard Pretix image.

Thanks

**Reply by raphaelm** [reply]
You can set NUM_WORKERS as an environment variable. concurrency is currently not accessible in the docker image without changes

**Reply by mklemme1** [reply]
Thanks!
Setting `NUM_WORKERS=1` reduces the memory footprint of Pretix from 2 GB to 770 MB.
We have a very small instance, I therefore think one should be enough.
